packages/types/src/api-key/service.ts

```diff
@@ -15,11 +15,31 @@ export interface IApiKeyModuleService extends IModuleService {
 
   /**
    * Update an api key
+   * @param selector
    * @param data
    * @param sharedContext
    */
-  update(data: UpdateApiKeyDTO[], sharedContext?: Context): Promise<ApiKeyDTO[]>
-  update(data: UpdateApiKeyDTO, sharedContext?: Context): Promise<ApiKeyDTO>
+  update(
+    selector: FilterableApiKeyProps,
+    data: Omit<UpdateApiKeyDTO, "id">,
+    sharedContext?: Context
+  ): Promise<ApiKeyDTO[]>
+  /**
+   * Update an api key
+   * @param id
+   * @param data
+   * @param sharedContext
+   */
+  update(
+    id: string,
+    data: Omit<UpdateApiKeyDTO, "id">,
+    sharedContext?: Context
+  ): Promise<ApiKeyDTO>
+  /**
+   * Update an api key
+   * @param data
+   */
+  update(data: UpdateApiKeyDTO[]): Promise<ApiKeyDTO[]>
```

Comment:
q: missing shared context, also, is the selector based approach needed here?

Reply:
also, it would be better to have the shared context at always the same position as the index of the shared context is stored on the class through the decorator and then used to identify where to put it. Here the second argument of the method could then be data or shared context and can create issues. Maybe an huddle would be easier if you want to discuss it

Reply:
@adrien2p similar to above, I just followed the API pattern of the region, as I assumed that's how we want all modules to behave.

I think we just need to standardize how all modules behave basically, as I am not sure what the "expected" public API for each module is. 

I would suggest we do the discussion separately from the PR, and we go with this one as it is, and then we modify all modules at once that don't follow the convention.

Reply:
I do agree on the standardization for sure 👍 though, you will have an issue with the context coming from a workflow for example

Reply:
@adrien2p let me merge this and I'll update it in the follow-up PR where I enforce the usage of the authentication, since this already built and passed 😄